sktime/forecasting/adapters/_autots.py

```diff
@@ -0,0 +1,449 @@
+        model_list (str): The list of models to use. See docs for details.
+        frequency (str): The frequency of the time series.
+        prediction_interval (float): The prediction interval for the forecast.
+        max_generations (int): The maximum number of
+            generations for the genetic algorithm.
+        no_negatives (bool): Whether negative values are allowed in the forecast.
+        constraint (float): The constraint on the forecast values.
+        ensemble (str): The ensemble method to use.
+        initial_template (str): The initial template to use for the forecast.
+        random_seed (int): The random seed for reproducibility.
+        holiday_country (str): The country for holiday effects.
+        subset (int): The subset of the data to use.
+        aggfunc (str): The aggregation function to use.
+        na_tolerance (float): The tolerance for missing values.
+        metric_weighting (dict): The weights for different forecast evaluation metrics.
+        drop_most_recent (int): The number of most recent data points to drop.
+        drop_data_older_than_periods (int): The threshold for dropping old data points.
+        transformer_list (dict): The list of transformers to use.
+        transformer_max_depth (int): The maximum depth of the transformers.
+        models_mode (str): The mode for selecting models.
+        num_validations (int): The number of validations to perform.
+        models_to_validate (float): The fraction of models to validate.
+        max_per_model_class (int): The maximum number of models per class.
+        validation_method (str): The method for validation.
+        min_allowed_train_percent (float): The minimum percentage of
+            data allowed for training.
+        remove_leading_zeroes (bool): Whether to remove leading zeroes from the data.
+        prefill_na (str): The method for prefilling missing values.
+        introduce_na (bool): Whether to introduce missing values to the data.
+        preclean (dict): The parameters for data pre-cleaning.
+        model_interrupt (bool): Whether the model can be interrupted.
+        generation_timeout (int): The timeout for each generation.
+        current_model_file (str): The file containing the current model.
+        verbose (int): The verbosity level.
+        n_jobs (int): The number of jobs to run in parallel.
+        model_name: (str): The name of the model.
+            NOTE: Overwrites the model_list parameter. For using only one model.
+    """
+
+    _tags = {
+        "scitype:y": "univariate",
+        "authors": [
+            "MBristle",
+        ],
+        # "maintainers": ["MBristle"],
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "ignores-exogeneous-X": True,
+        "capability:insample": False,
+        "capability:pred_int:insample": False,
+        "requires-fh-in-fit": True,
+        "reserved_params": ["fh_"],
+        "python_dependencies": "autots",
```

Comment:
Since this is missing from `pyproject.toml` currently, CI will not test anything. You can add a new entry in `all_extras`, `all_extras_pandas2` and `forecasting`, and let's see what happens.

Reply:
Related question: what all are dependencies? `autots` has a few optional dependencies, do this adapter work without them?

For example, can `holiday_country` work without `holidays`?

Reply:
hm, if you have dependencies that are conditional on parameters, the "clean" way would be to dynamically set the dependencies tag before `super().__init__()`, depending on the parameters.

This seems very tedious though. How doe `autots` manage its dependencies? Surely it tells the user if they need sth - or, if everything is just a core dependency, then there is also nothing to do.

Reply:
I have not installed any dependency other than autots in the venv for development. holidays get picked up and passed to the holiday package in case it is used (I guess, e.g., in Prophet). So in my opinion there are no other dependencies than the core dependencies of autots (added them explicitly). If a package of a model that should be used in autots is not installed I see this as the responsibility of autots and not the adapter.

Reply:
makes sense, and is also easier to maintain